twister: test: pytest -s scripts/tests/twister/test_runner.py fails

**Describe the bug**
pytest -s scripts/tests/twister/test_runner.py fails


**To Reproduce**
Steps to reproduce the behavior:
1.  run below
 pytest -s scripts/tests/twister/test_runner.py

**Expected behavior**
test pass

**Impact**
CI

**Logs and console output**

```
scripts/pylib/twister/twisterlib/runner.py:268: TypeError
---------------------------------------------------------------------------------------------- Captured log call -----------------------------------------------------------------------------------------------DEBUG    twister:runner.py:247 Building <Mock name='mock.testsuite.source_dir' id='140469417446368'> for <Mock name='mock.platform.name' id='140469417456064'>
=========================================================================================== short test summary info ============================================================================================FAILED scripts/tests/twister/test_runner.py::test_projectbuilder_gather_metrics[none] - TypeError: cannot unpack non-iterable Mock object
FAILED scripts/tests/twister/test_runner.py::test_projectbuilder_gather_metrics[size_report] - TypeError: cannot unpack non-iterable Mock object
FAILED scripts/tests/twister/test_runner.py::test_projectbuilder_gather_metrics[cmake] - TypeError: cannot unpack non-iterable Mock object
FAILED scripts/tests/twister/test_runner.py::test_projectbuilder_gather_metrics[size_report+cmake] - TypeError: cannot unpack non-iterable Mock object
======================================================================================== 4 failed, 128 passed in 1.91s =========================================================================================ubuntu@ubuntu-OptiPlex-7050:/home/shared/disk/zephyr_project/zephyr_test/zephyr$ 

```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: zephyr-v3.5.0-770-gaa3e79a88a

git bisect find this below commit introduces this case.

```
commit 2235a8253bdda4642e863849351576701931fac4
Author: Lukasz Mrugala <lukaszx.mrugala@intel.com>
Date:   Thu Oct 5 11:47:11 2023 +0200

    scripts: tests: twister: runner test expansion
    
    Creates a few dozen new tests that cover every runner.py method.
    99% coverage (All statements save for one) is achieved.
    
    Signed-off-by: Lukasz Mrugala <lukaszx.mrugala@intel.com>
```

--- a/scripts/tests/twister/test_runner.py
+++ b/scripts/tests/twister/test_runner.py
@@ -2235,6 +2235,7 @@ def test_projectbuilder_gather_metrics(
 
     pb = ProjectBuilder(instance_mock, env_mock, mocked_jobserver)
     pb.options.enable_size_report = enable_size_report
+    pb.options.create_rom_ram_report = False
     pb.options.cmake_only = cmake_only
     pb.calc_size = mock.Mock()
 

